app/components/contacts/SearchableList.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import styled from 'styled-components';
+import { smColors } from '/vars';
+import type { Contact } from '/types';
+
+const Wrapper = styled.div`
+  margin-bottom: 24px;
+`;
+
+const TitleWrapper = styled.div`
+  height: 32px;
+  border-bottom: 1px solid ${smColors.borderGray};
+`;
+
+const ItemsContainer = styled.div`
+  max-height: 240px;
+  overflow-y: scroll;
+`;
+
+const Text = styled.span`
+  font-size: 16px;
+  font-weight: normal;
+  line-height: 22px;
+`;
+
+const BoldText = styled(Text)`
+  font-weight: bold;
+`;
+
+const Title = styled.span`
+  font-size: 14px;
+  font-weight: bold;
+  line-height: 19px;
+  color: ${smColors.darkGray};
+`;
+
+const AddressRow = styled.div`
+  height: 76px;
+  border-bottom: 1px solid ${smColors.borderGray};
+  padding-bottom: 12px;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  padding: 12px 0;
+`;
+
+const ActionLink = styled(Text)`
+  color: ${smColors.orange};
+  padding-left: 12px;
+  font-size: 14px;
+  cursor: pointer;
+  &:hover {
+    opacity: 0.8;
+  }
+  &:active {
+    opacity: 0.6;
+  }
+`;
+
+type Props = {
+  title: string,
+  searchPhrase: string,
+  list: Contact[],
+  onSave: (entry: Contact) => {}
+};
+
+const SearchableList = ({ searchPhrase, list, title, onSave }: Props) => {
+  const filteredList = list.filter((listItem: Contact) => {
```

Comment:
instead of making double pass through array (filter and then map in render) just return null in render func for items not matching search criteria

Reply:
Much better! Changed, Thanks.